content/docs/rfcs/17/README.md

```diff
@@ -106,24 +109,26 @@ The reason is that using an old root can allow inference about the index of the
 Upon the receipt of a PubSub message via [`11/WAKU2-RELAY`](/spec/11) protocol, the routing peer parses the `data` field as a `WakuMessage` and gets access to the `RateLimitProof` field.  
 The peer then validates the `RateLimitProof`  as explained next.
 
-**Epoch Validation**
+### Epoch Validation
 If the `epoch` attached to the message is more than `max_epoch_gap` apart from the routing peer's current `epoch` then the message is discarded and considered invalid.
 This is to prevent a newly registered peer from spamming the system by messaging for all the past epochs. 
 `max_epoch_gap` is a system parameter for which we provide some recommendations in section [Recommended System Parameters](#recommended-system-parameters).
 
-**Merkle Root Validation**
+### Merkle Root Validation
 The routing peers MUST check whether the provided Merkle root in the `RateLimitProof` is valid.
 It can do so by maintaining a local set of valid Merkle roots, which consist of `acceptable_root_window_size` past roots.
+These roots refer to the final state of the Merkle tree after a whole block consisting of group changes are processed.
+The Merkle roots are updated on a per-block instead of a per-event basis since a large number of group changes in a single block may fully populate the acceptable window of roots, effectively preventing some peers from sending messages with a root that refers to the state of the Merkle tree at a previous block. 
```

Comment:
`short period of time` would more succinctly be `single block`, right? 

How does this sound - 
```
The Merkle roots are updated on a per-block instead of a per-event basis since a large number of group changes in a single block may fully replace the acceptable window of roots, effectively leading to some peers from sending invalid messages with a root that refers to the state of the Merkle tree at a previous block, even if it appears valid to them. 
```

Reply:
oh! removed the `from`. Will try to split it

Reply:
Tried to address it in f348802, wdyt?